src/compiler/utilities.ts

```diff
@@ -2452,17 +2452,18 @@ namespace ts {
     }
 
     export function getHostSignatureFromJSDoc(node: Node): SignatureDeclaration | undefined {
-        return getHostSignatureFromJSDocHost(getJSDocHost(node));
+        const host = findJSDocHost(getJSDocHost(node));
+        return host && isFunctionLike(host) ? host : undefined;
     }
 
-    export function getHostSignatureFromJSDocHost(host: HasJSDoc): SignatureDeclaration | undefined {
+    export function findJSDocHost(host: HasJSDoc): Node | undefined {

```

Comment:
Uhhhhm, how about `getJSDocHost` and `reallyActuallyGetJSDocHostFromWrongHost` 🤔 

`getTargetJSDocHost`? `getTargetNodeOfJSDoc`?

Reply:
`getEffectiveJSDocHost` ... is clunky but kind of agrees with our existing, similarly clunky, scheme.